### PR TITLE
DOCSP-30345: Update Kotlin SDK's Update page

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Update existing [Update](https://www.mongodb.com/docs/realm/sdk/kotlin/realm-database/crud/update/) page with missing content and improved code examples using updated object models

### Jira

- https://jira.mongodb.org/browse/DOCSP-30345

### Staged Changes

- [Update - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-30345-update-page/sdk/kotlin/realm-database/crud/update/) - add missing sections and update code examples
- [Delete - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-30345-update-page/sdk/kotlin/realm-database/crud/delete/) - small updates
- [Supported Types - Kotlin SDK ](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-30345-update-page/sdk/kotlin/realm-database/schemas/supported-types/)- small updates

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/realm-database/crud/delete.txt [hunk 1]
```diff
@@ -294,16 +294,20 @@ until you manually delete them. Alternatively, deleting a Realm object from a
...
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _kotlin-remove-realmlist-elements:
+
+Remove Elements from a RealmList
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-You can remove one or more items from a ``RealmList`` at a time:
+You can remove one or more elements from a 
+`RealmList <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-list/index.html>`__  
+at a time:
```
Comment (krollins-mdb) [inline source/sdk/kotlin/realm-database/crud/delete.txt:304]: The "at a time" feels odd here. What do you think about using "in a single transaction" or something similar?

After thinking about it, what feels odd is "at a time" being at the end of the sentence - it's too far from its subject. I think this feels better:
```suggestion
You can remove one or more elements at a time from a 
`RealmList <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-list/index.html>`__ :
```

Comment (krollins-mdb) [reply source/sdk/kotlin/realm-database/crud/delete.txt:304]: Whatever you go with, if changes are made, they should be applied in other places where this construction is.

**Review by krollins-mdb — APPROVED** [review]
Just a couple minor suggestions! Nice work, @cbullinger!

File: source/sdk/kotlin/realm-database/crud/delete.txt (hunk 1 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database/crud/delete.txt:304]: Updated throughout

**Pull request merged**